pkg/config/process.go

```diff
@@ -70,3 +94,48 @@ func setupProcesses(config Config) {
 	)
 	procBindEnvAndSetDefault(config, "process_config.process_discovery.interval", 4*time.Hour)
 }
+
+var displayProcConfigEnabledWarningOnce sync.Once
+
+// displayProcConfigEnabledDeprecationWarning displays a deprecation warning for process_config.enabled only once.
+// For testing purposes, it rethrows the warning.
+func displayProcConfigEnabledDeprecationWarning() (warning error) {
+	displayProcConfigEnabledWarningOnce.Do(func() {
+		warning = log.Warn("process_config.enabled is deprecated, use process_config.container_collection.enabled" +
+			" and process_config.process_collection.enabled instead")
+	})
+	return
+}
+
+// GetContainerCollectionEnabled retrieves the value of process_config.container_collection.enabled.
+// If process_config.enabled is set, we display a deprecation warning and use that value instead.
+func GetContainerCollectionEnabled(config Config) bool {
+	if config.IsSet("process_config.enabled") {
+		_ = displayProcConfigEnabledDeprecationWarning()
+
+		procConfigEnabled := strings.ToLower(config.GetString("process_config.enabled"))
+		if procConfigEnabled == "disabled" {
+			return false
+		}
+		result, _ := strconv.ParseBool(procConfigEnabled)
+		return !result
+	}
+	return config.GetBool("process_config.container_collection.enabled") &&
+		!config.GetBool("process_config.process_collection.enabled") // Process Collection already collects containers
```

Comment:
It would be best to have this just return `config.GetBool("process_config.container_collection.enabled")` at this point and handle instantiating `Container` check with this logic here. My thinking about this - we may want to use this value in various config reporting places (like agent status) and it will be difficult to understand that "container collection is not enabled" but "process collection is enabled so it collects containers too", it is best to show "container collection is enabled" at that point and handle internal detail intelligently. Thoughts?

Reply:
That makes sense. To be honest I was a bit conflicted as to whether or not I should encapsulate this logic!

I think that makes sense to ignore the value of `process_collection.enabled` in case we want to use it in extra places. It would certainly be confusing if in some places you need to use `config.GetBool("process_config.container_collection.enabled"` and `GetContainerCollectionEnabled` in others. I'll go ahead and change the behavior of this